cmd/clusterctl/cmd/completion.go

```diff
@@ -73,20 +62,29 @@ var (
 		printf "\n# clusterctl shell completion\nsource '$HOME/.kube/clusterctl_completion.bash.inc'\n" >> $HOME/.bash_profile
 		source $HOME/.bash_profile
 
-		# Load the clusterctl completion code for zsh[1] into the current shell
-		source <(clusterctl completion zsh)`)
+		Zsh:
+		# If shell completion is not already enabled in your environment you will need
+		# to enable it.  You can execute the following once:
+		echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+		# To load completions for each session, execute once:
+		clusterctl completion zsh > "${fpath[1]}/_clusterctl"
```

Comment:
I am getting a permission error when running this command, 
>zsh: permission denied: /usr/local/share/zsh/site-functions/_clusterctl

Not sure if only it is system-specific.

Reply:
It depends on the system.